File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    result = 0
    while storey:
        stor = storey % 10
        if stor > 5:
            result += 10-stor
            storey += 10
        elif stor < 5:
            result += stor
        else:
            if (storey // 10) % 10 > 4:
                storey += 10
            result += stor        
        storey //= 10

    return result

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    let answer = 0;
    
    while (storey) {
        let stor = storey % 10
        if (stor > 5) {
            storey += 10
            answer += (10-stor)
        } else if (stor < 5) {
            answer += stor
        } else {
            if (Math.floor(storey / 10) % 10 > 4) {
                storey += 10
            }
            answer += stor
        }
        storey = Math.floor(storey / 10)
    }
    return answer;
}